frontend/src/components/appbar/AppBar.tsx

```diff
@@ -94,13 +96,33 @@ export default function AppBarComp(props: AppBarProps) {
           </Typography>
           <IconButton
             aria-label="User's profile avatar"
-            color="inherit"
             className={classes.accountIcon}
+            color="inherit"
           >
-            <AccountCircleIcon fontSize="large"/>
+            <UserAvatarImage />
           </IconButton>
         </Toolbar>
       </AppBar>
-    </div>
+    </>
   );
 }
+
+/**
+ * Extracts the user profile image contained in the OAuth token.
+ * If the user token is not defined, returns a stock account icon.
+ */
+export function UserAvatarImage() {
+  const classes = useStyles();
+
+  const contextServices = React.useContext(UserLoginStatusContext);
+  const loginStatusHandlerService = contextServices.loginStatusHandlerService;
+
+  const token = loginStatusHandlerService.getUserToken();
+  if (token) {
+    const parsedToken = JSON.parse(atob(token.split('.')[1]));
+    const profileImg = parsedToken.picture;
```

Comment:
As far as I know this isn't guaranteed to exist, so also fall back to `<AccountCircleIcon />` if the token cannot be parsed or if it has an empty 'picture' field.